backend/ventserver/integration/_trio.py

```diff
@@ -231,6 +232,11 @@ async def process_io_persistently(
     async with push_endpoint:
         while True:
             await io_endpoint.persistently_open(nursery=nursery)
+            if isinstance(io_endpoint, websocket_io.Driver):
+                protocol.receive.frontend_connected =\
```

Comment:
For asynchronous execution correctness, we should only pass data to/from the protocol through `protocol.receive.input`/`protocol.receive.output`/`protocol.send.input`/`protocol.send.output`.